pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go

```diff
@@ -86,6 +86,10 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	if workloadType == util.ReferWorkload {
 		def.WorkloadDefName = componentDefinition.Spec.Workload.Type
 	}
+	if workloadType == util.HELMDef {
+		def.Helm = componentDefinition.Spec.Schematic.HELM
+		def.ComponentDefinition = componentDefinition
+	}
```

Comment:
Currently just make minimal changes in CompDef controller in this PR. 
I would like to refactor it latter.